#112 add View interface and implement it in the Console class

kortov opened the pull request:
#112

kortov committed:
#112 add View interface and implement it in the Console class

- add View interface and implement
- implement View in the Console class
- add tests

AVM82 commented on src/main/java/juja/sqlcmd/view/Console.java:
```diff
@@ -0,0 +1,22 @@
+import java.util.Scanner;
+
+public class Console implements View {
+    Scanner scanner;
```
Perhaps we can make the Scanner scanner class field private

kortov replied:
Yep)

AVM82 reviewed (commented):
Not sure, but it seems to me that there is no need to test the Console class at the bottom stage

kortov committed:
#112 Minor fixes: rename variables and narrow fields access

- renamed field console in the ConsoleTest
- modified test strings in the ConsoleTest
- narrowed the access level of the View field

Shtramak commented on src/main/java/juja/sqlcmd/view/Console.java:
```diff
@@ -3,7 +3,7 @@
 
 public class Console implements View {
-    Scanner scanner;
+    private Scanner scanner;
```
Why to have a Scanner as a field? You don't return it anywhere and will never reuse somewhere else... I think it's redundant

Shtramak commented on src/main/java/juja/sqlcmd/view/Console.java:
```diff
@@ -3,7 +3,7 @@
-    Scanner scanner;
+    private Scanner scanner;
 
     public Console() {
```
If you will delete a field then empty constructor will be enough. Also pay your attention that you initialize scanner field twice: once in a constructor and then in read() method

kortov committed:
#112 Refactored (added DI and more abstractions

- Added DI for input and output streams in the Console class
- refactored tests, added helping method for InputStream

AVM82 commented on src/main/java/juja/sqlcmd/view/Console.java:
```diff
@@ -0,0 +1,35 @@
+    public void write(String message) {
+        try {
+            String modifiedToMimicUserInput = message + System.lineSeparator();
+            outputStream.write(modifiedToMimicUserInput.getBytes());
```
You can write this in one line

AVM82 reviewed (approved):
good
